apps/readest-app/src/app/reader/utils/iframeEventHandlers.ts

```diff
@@ -130,3 +130,36 @@ export const handleClick = (bookKey: string, event: MouseEvent) => {
     }
   }, doubleClickThreshold);
 };
+
+const handleTouchEv = (bookKey: string, event: TouchEvent, type: string) => {
+  const touch = event.targetTouches[0];
+  const touches = [];
+  if (touch) {
+    touches.push({
+      clientX: touch.clientX,
+      clientY: touch.clientY,
+      screenX: touch.screenX,
+      screenY: touch.screenY,
+    });
+  }
+  window.postMessage(
```

Comment:
it would be good to figure out a better way to propagate events to make the event handling logic more efficient, right now every handler need to subscribe and filter all messages.